sdk/storage/storage-blob/review/generated/src/models/index.d.ts

```diff
@@ -0,0 +1,6713 @@
+     */
+    allowedOrigins: string;
+    /**
+     * The methods (HTTP request verbs) that the origin domain may use for a CORS request. (comma
+     * separated)
+     */
+    allowedMethods: string;
+    /**
+     * the request headers that the origin domain may specify on the CORS request.
+     */
+    allowedHeaders: string;
+    /**
+     * The response headers that may be sent in the response to the CORS request and exposed by the
+     * browser to the request issuer
+     */
+    exposedHeaders: string;
+    /**
+     * The maximum amount time that a browser should cache the preflight OPTIONS request.
+     */
+    maxAgeInSeconds: number;
+}
+/**
+ * Geo-Replication information for the Secondary Storage Service
+ */
+export interface GeoReplication {
+    /**
+     * The status of the secondary location. Possible values include: 'live', 'bootstrap',
+     * 'unavailable'
+     */
+    status: GeoReplicationStatusType;
+    /**
+     * A GMT date/time value, to the second. All primary writes preceding this value are guaranteed
+     * to be available for read operations at the secondary. Primary writes after this point in time
+     * may or may not be available for reads.
+     */
+    lastSyncTime: Date;
+}
+/**
+ * the retention policy which determines how long the associated data should persist
+ */
+export interface RetentionPolicy {
+    /**
+     * Indicates whether a retention policy is enabled for the storage service
+     */
+    enabled: boolean;
+    /**
+     * Indicates the number of days that metrics or logging or soft-deleted data should be retained.
+     * All data older than this value will be deleted
+     */
+    days?: number;
+}
+/**
+ * Azure Analytics Logging settings.
+ */
+export interface Logging {
+    /**
+     * The version of Storage Analytics to configure.
+     */
+    version: string;
+    /**
+     * Indicates whether all delete requests should be logged.
+     */
+    deleteProperty: boolean;
+    /**
+     * Indicates whether all read requests should be logged.
+     */
+    read: boolean;
+    /**
+     * Indicates whether all write requests should be logged.
+     */
+    write: boolean;
+    retentionPolicy: RetentionPolicy;
+}
+/**
+ * a summary of request statistics grouped by API in hour or minute aggregates for blobs
+ */
+export interface Metrics {
+    /**
+     * The version of Storage Analytics to configure.
+     */
+    version?: string;
+    /**
+     * Indicates whether metrics are enabled for the Blob service.
+     */
+    enabled: boolean;
+    /**
+     * Indicates whether metrics should generate summary statistics for called API operations.
+     */
+    includeAPIs?: boolean;
+    retentionPolicy?: RetentionPolicy;
+}
+/**
+ * An interface representing PageRange.
+ */
+export interface PageRange {
+    start: number;
+    end: number;
+}
+/**
+ * An interface representing ClearRange.
+ */
+export interface ClearRange {
```

Comment:
It's very strange to have both Page and Clear range. We should have one range type in azure-core

Reply:
There is an HTTP Range where we could have an Azure Core type, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range

Reply:
We could unify this in the generator, else we can unify above that layer.

Let's prefer to fix in swagger if we can.